src/ext/java/org/opentripplanner/ext/fares/impl/OrcaFareService.java

```diff
@@ -217,62 +246,65 @@ private static RideType classify(Route route, String tripId) {
   }
 
   /**
-   * Define which discount fare should be applied based on the fare type. If the ride type is unknown the discount
-   * fare can not be applied, use the default fare.
+   * Define which discount fare should be applied based on the fare type. If the ride type is
+   * unknown the discount fare can not be applied, use the default fare.
    */
-  private Money getLegFare(FareType fareType, RideType rideType, Money defaultFare, Leg leg) {
+  private Optional<Money> getLegFare(
```

Comment:
This method returns an Optional now but it appears to never return `Optional.empty`. Am I right?

If so, then you should simply wrap the call site in `Optional.of`.

Reply:
It returns `Optional.empty` a few lines down in the if statement.